lib/util/mime_utils.js

```diff
@@ -42,9 +45,15 @@ shaka.util.MimeUtils = class {
    */
   static getFullOrConvertedType(mimeType, codecs, contentType) {
     const fullMimeType = shaka.util.MimeUtils.getFullType(mimeType, codecs);
+    const ContentType = shaka.util.ManifestParserUtils.ContentType;
 
-    if (!shaka.dependencies.muxjs() ||
-        !shaka.media.Transmuxer.isTsContainer(fullMimeType)) {
+    if (shaka.media.Transmuxer.isTsContainer(fullMimeType)) {
```

Comment:
Hm. It took me a few tries to get the flow of this logic. I think this might be clearer if returning `fullMimeType` was the fall-through case, not `convertTsCodecs`. Maybe plus some comments to explain why the audio converting is where it is in the logic.

Something like:
```js
    if (shaka.media.Transmuxer.isTsContainer(fullMimeType)) {
      if (shaka.dependencies.muxjs()) {
        return shaka.media.Transmuxer.convertTsCodecs(
            contentType, fullMimeType);
      }
    } else if (contentType == ContentType.AUDIO) {
      // video/mp2t is the correct mime type for TS audio, so only replace the
      // word "video" with "audio" for non-TS audio content.
      return fullMimeType.replace('video', 'audio');
    }
    return fullMimeType;
```

Reply:
Done.